tests/test_file.py

```diff
@@ -263,3 +267,78 @@ def test_add_array_to_group_udf_non_existing_group_raises_value_error(self):
                 config=client.Ctx().config().dict(),
                 verbose=True,
             )
+
+
+class TestFileIndexing(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Setup test files, group and destinations once before the file tests start."""
+        cls.input_file_location = "s3://tiledb-unittest/groups/file_indexing_test_files"
```

Comment:
To avoid having a hard dependency on files that it’s impossible for users who aren’t us to access, would it make sense to do something like:

 1. Put these files in the testdata folder
 2. Copy them to a new directory in cloud storage when the test starts
 3. Add a way to override this path (environment variable?)

?

This would also let us do something like putting all the data/results in a path that looks like

<code><var>cloud storage base</var>/run-<var>timestamp</var>-<var>random suffix</var>/...</code>

to make automatic cleanup easier if the test crashes or is interrupted in the middle and can't clean itself up.

Reply:
I have tried that for the ingestion tests, but the CI runner has no VFS access so we have abandoned that idea for now.

Reply:
> We have tried that for the ingestion tests, but the CI runner has no VFS access so we have abandoned that idea for now.

We could enable this. Did we consider adding those permissions? I agree with @thetorpedodog here that making this reproducible would go a long way to future support-ability. It doesn't have to block this PR but we should consider an immediate follow up. `s3://tiledb-unittest` as a bucket also need followups, @antalakas lets discuss offline.

Reply:
Sure I am onboard for that. Will also make the ingestion&indexing testing less flaky.
I will add a followup for that in the ticket.